### PR TITLE
image,seed/seedwriter: switch image to use seedwriter.Writer

### PR DESCRIPTION
**Pull request by pedronis**
For this introduce Writer.BootSnaps|UnassertedSnaps|Warnings.

### CODE REVIEW

**Comment by pedronis** [comment]
based on #7541

File: seed/seedwriter/writer.go [hunk 1]
```diff
@@ -864,6 +878,12 @@ func (w *Writer) snapDecl(sn *SeedSnap) (*asserts.SnapDeclaration, error) {
 	return nil, fmt.Errorf("internal error: snap %q has no snap-declaration set", sn.SnapName())
 }
 
+// Warnings returns the warning messages produced so far. No warnings
+// should be generated after Downloaded signaled complete.
+func (w *Writer) Warnings() []string {
```
Comment (stolowski) [inline seed/seedwriter/writer.go]: I reckon this is unrelated to our warnings API; should we name it differently to avoid any confusion?

Comment (pedronis) [reply seed/seedwriter/writer.go]: it's unrelated, but at some point we have to trust people to be able to do context, it might be better than to invent more weird names for things

File: seed/seedwriter/writer.go [hunk 2]
```diff
@@ -921,3 +941,58 @@ func (w *Writer) WriteMeta() error {
...
+
+		}
+	}
+	return bootSnaps, nil
+}
+
+// UnassertedSnaps returns references for all unasserted snaps in the seed.
+// It can be invoked only after Downloaded returns complete ==
+// true.
+func (w *Writer) UnassertedSnaps() ([]naming.SnapRef, error) {
```
Comment (stolowski) [inline seed/seedwriter/writer.go]: Small wondering (without looking at the rest of the code) if we could have consistency here wrt returned types in `BoostSnaps()` and `UnassertedSnaps` (i.e. []*SeedSnap vs []naming.SnapRef). Is it possible to infer one from the other and use more generic one for both functions?

File: image/helpers.go [hunk 3]
```diff
@@ -209,8 +209,10 @@ func NewToolingStore() (*ToolingStore, error) {
 
 // DownloadOptions carries options for downloading snaps plus assertions.
 type DownloadOptions struct {
+	TargetDir  string
+	TargetFunc func(*snap.Info) (string, error)
```
Comment (stolowski) [inline image/helpers.go]: Could you please document what are the semantics of TargetDir and TargetFunc and how do they play together (it's not immediately obvious looking at the `String()`).

File: seed/seedwriter/writer.go (hunk 2 above)
Comment (pedronis) [reply seed/seedwriter/writer.go]: it's an encapsulation decision, one return type says the code expects you to need and care about the details,  the other expects that knowing the "names" is enough

File: image/helpers.go (hunk 3 above)
Comment (mvo5) [reply image/helpers.go]: Without having read further I wonder if TargetFunc() refers to a target filename or a target dir, maybe this could be made a bit more obvious?

File: image/helpers_test.go [hunk 4]
```diff
@@ -31,58 +31,69 @@ import (
...
-func (s *imageSuite) TestDownloadOptionsString(c *check.C) {
-	for opts, str := range map[image.DownloadOptions]string{
-		{LeavePartialOnError: true}: "",
-		{}:                     "",
-		{TargetDir: "/foo"}:    `in "/foo"`,
-		{Basename: "foo"}:      `to "foo.snap"`,
-		{Channel: "foo"}:       `from channel "foo"`,
-		{Revision: snap.R(42)}: `(42)`,
-		{
+func (s *imageSuite) TestDownloadpOtionsString(c *check.C) {
```
Comment (mvo5) [inline image/helpers_test.go]: Typo: "Otions" <- should be "Options"
